palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py

```diff
@@ -0,0 +1,301 @@
+        }
+
+        return output_object
+
+    def isolate_endpoint(self, endpoint, isolation_state):
+        endpoint_info = self.get_endpoint_information(endpoint)
+        endpoints = endpoint_info.get("endpoints")
+        if not endpoints:
+            self.logger.error(f"No matching endpoint found. Endpoint: {endpoint}")
+            raise PluginException(
+                cause="Endpoint not found",
+                assistance=f"The endpoint {endpoint} was not found. Please check your input and try again",
+            )
+
+        num_endpoints = len(endpoints)
+        self.logger.info(f"Number of endpoints found: {num_endpoints}")
+
+        if num_endpoints > 1:
+            self.logger.info(f"Taking isolation action on multiple endpoints.")
+            return self._isolate_multiple_endpoints(endpoints, isolation_state)
+        else:
+            self.logger.info(f"Taking isolation action on a single endpoint.")
+            return self._isolate_endpoint(endpoints, isolation_state)
+
+    def get_incidents(self, from_time, to_time, time_field="creation_time"):
+        endpoint = "/public_api/v1/incidents/get_incidents/"
+
+        batch_size = 100
+        search_from = 0
+        search_to = search_from + batch_size
+
+        # Request incidents in ascending order so that we get the oldest events first.
+        post_body = {
+            "request_data": {
+                "search_from": search_from,
+                "search_to": search_to,
+                "sort": {"field": time_field, "keyword": "asc"},
+            }
+        }
+
+        # If time constraints have been provided for the request, add them to the post body
+        if from_time is not None and to_time is not None:
+            post_body["request_data"]["filters"] = [
+                {"field": time_field, "operator": "gte", "value": from_time},
+                {"field": time_field, "operator": "lte", "value": to_time},
+            ]
+
+        done = False
+        all_incidents = []
+
+        # Keep paging through and requesting incidents until we have requested all incidents which match our query
+        while not done:
+            resp_json = self._post_to_api(endpoint, post_body)
+            if resp_json is not None:
+                total_count = resp_json.get("reply", {}).get("total_count", -1)
+                all_incidents.extend(resp_json.get("reply", {}).get("incidents", []))
+                # If the number of incidents we have received so far is greater than or equal to the total number of
+                # incidents which match the query, then we can stop paging.
+                if len(all_incidents) >= total_count or total_count < 0:
+                    done = True
+
+                # Update the indices of search_from and search_to to we can request the next page
+                search_from = search_from + batch_size
+                search_to = search_to + batch_size if search_to + batch_size < total_count else total_count
```

Comment:
Something like this would be more readable IMO
```suggestion
                new_to = search_to + batch_size
                search_to = new_to if new_to < total_count else total_count
```